File: src/main/java/me/taubsie/carrylogs/application/classes/PurgeData.java
package me.taubsie.carrylogs.application.classes;

import me.taubsie.carrylogs.application.enums.RoleConversion;

import java.util.List;

/**
 * This record holds the purge data used for a single user to allow easy querying.
 *
 * @param serverId The server on which the purge took place.
 * @param userId The user to purge.
 * @param rolesToRemove The roles which are set to be removed.
 * @param score The score which the user had.
 * @param purgeType The type of purge.
 * @param purgeThreshold The threshold which the purge had.
 */
public record PurgeData(long serverId, long userId, List<RoleConversion> rolesToRemove, Long score, String purgeType, long purgeThreshold) {
}

File: src/main/kotlin/net/dungeonhub/application/misc/PurgeData.kt
package net.dungeonhub.application.misc

import net.dungeonhub.model.discord_role.DiscordRoleModel
import net.dungeonhub.model.purge_type.PurgeTypeModel

/**
 * This record holds the purge data used for a single user to allow easy querying.
 *
 * @param userId         The user to purge.
 * @param rolesToRemove  The roles which are set to be removed.
 * @param score          The score which the user had.
 * @param purgeType      The type of purge.
 * @param purgeThreshold The threshold which the purge had.
 */
@JvmRecord
data class PurgeData(
    val userId: Long, val rolesToRemove: List<DiscordRoleModel>, val score: Long, val purgeType: PurgeTypeModel,
    val purgeThreshold: Long
) 